File: backend/routes/certificate.go
package routes

import (
	"app/controllers/certificate"
	"app/middlewares"

	"github.com/gin-gonic/gin"
)

func InitCertificate(r *gin.RouterGroup, e *gin.Engine) {
	const baseUrl = "/:id/certificates"

	r.GET(baseUrl + "/", middlewares.CheckWorkspace, certificate.Index)
	r.POST(baseUrl + "/", middlewares.CheckWorkspace, certificate.Create)
	r.PUT(baseUrl + "/:certificateId", middlewares.CheckWorkspace, certificate.Update)
	r.DELETE(baseUrl + "/:certificateId", middlewares.CheckWorkspace, certificate.Delete)
	r.PUT(baseUrl + "/:certificateId/renew", middlewares.CheckWorkspace, certificate.Renew)
}

File: frontend/src/api/certificate.ts
import http from './_http';

export default {
  index (workspaceId: number) {
    return http.get(`/workspaces/${workspaceId}/certificates/`);
  },
  create (workspaceId: number, data: any) {
    return http.post(`/workspaces/${workspaceId}/certificates/`, data);
  },
  update (workspaceId: number, id: number, data: any) {
    return http.put(`/workspaces/${workspaceId}/certificates/${id}`, data);
  },
  delete (workspaceId: number, id: number) {
    return http.delete(`/workspaces/${workspaceId}/certificates/${id}`);
  },
  renew (workspaceId: number, id: number) {
    return http.put(`/workspaces/${workspaceId}/certificates/${id}/renew`);
  },
}